scanners/reactnative-expo/reactnative-expo.go

```diff
@@ -0,0 +1,706 @@
+	"github.com/bitrise-io/go-utils/command"
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-io/go-utils/pathutil"
+	yaml "gopkg.in/yaml.v2"
+)
+
+// Name ...
+const Name = "react-native-expo"
+
+const (
+	workDirInputKey = "workdir"
+
+	gradleFileInputKey    = "gradle_file"
+	gradleFileInputEnvKey = "GRADLE_BUILD_FILE_PATH"
+	gradleFileInputTitle  = "Path to the gradle file to use"
+
+	gradleTaskInputKey = "gradle_task"
+)
+
+// Scanner ...
+type Scanner struct {
+	searchDir          string
+	iosScanner         *ios.Scanner
+	androidScanner     *android.Scanner
+	reactnativeScanner *reactnative.Scanner
+	hasNPMTest         bool
+	packageJSONPth     string
+}
+
+// LineModification ...
+type LineModification struct {
+	lastTime    time.Time
+	answerCount int
+}
+
+// LastLineModification ...
+var LastLineModification atomic.Value
+
+// NewScanner ...
+func NewScanner() *Scanner {
+	return &Scanner{}
+}
+
+// Name ...
+func (Scanner) Name() string {
+	return Name
+}
+
+// DetectPlatform ...
+func (scanner *Scanner) DetectPlatform(searchDir string) (bool, error) {
+	scanner.searchDir = searchDir
+
+	log.TInfof("Collect package.json files")
+
+	packageJSONPths, err := reactnative.CollectPackageJSONFiles(searchDir)
+	if err != nil {
+		return false, err
+	}
+
+	if len(packageJSONPths) == 0 {
+		return false, nil
+	}
+
+	dependencyFound := false
+	for _, packageJSONPth := range packageJSONPths {
+		dependency := "expo"
+
+		var err error
+		dependencyFound, err = FindDependency(packageJSONPth, dependency)
+		if err != nil {
+			return false, err
+		}
+
+		log.Debugf("\n%s found: %t\n", dependency, dependencyFound)
+		if dependencyFound {
+			break
+		}
+	}
+
+	if !dependencyFound {
+		return false, nil
+	}
+
+	log.TPrintf("%d package.json file detected", len(packageJSONPths))
+
+	log.TInfof("Filter relevant package.json files")
+
+	fmt.Printf("Returning true for expo")
+	iosScanner := ios.NewScanner()
+	androidScanner := android.NewScanner()
+	for _, packageJSONPth := range packageJSONPths {
+		log.TPrintf("checking: %s", packageJSONPth)
+
+		projectDir := filepath.Dir(packageJSONPth)
+
+		iosDir := filepath.Join(projectDir, "ios")
+		if exist, err := pathutil.IsDirExists(iosDir); err != nil {
+			return false, err
+		} else if exist {
+			if detected, err := iosScanner.DetectPlatform(scanner.searchDir); err != nil {
+				return false, err
+			} else if detected {
+				return false, nil
+			}
+		}
+
+		androidDir := filepath.Join(projectDir, "android")
+		if exist, err := pathutil.IsDirExists(androidDir); err != nil {
+			return false, err
+		} else if exist {
+			if detected, err := androidScanner.DetectPlatform(scanner.searchDir); err != nil {
+				return false, err
+			} else if detected {
+				return false, nil
+			}
+		}
+	}
+
+	if err := ensureNodeModules(); err != nil {
+		log.Errorf("ERROR DURING INSTALLING NPM: %s", err)
+		return false, nil
+	}
+
+	if err := ejectProject(searchDir); err != nil {
```

Comment:
may we could pre-validate if there is an eject npm task or not (the npm tasks are listen in the package.json file)